src/analysis/calculus/cont_diff_def.lean

```diff
@@ -849,72 +852,87 @@ lemma norm_iterated_fderiv_within_fderiv_within {n : ℕ} (hs : unique_diff_on 
 by rw [iterated_fderiv_within_succ_eq_comp_right hs hx, linear_isometry_equiv.norm_map]
 
 @[simp] lemma iterated_fderiv_within_one_apply
-  (hs : unique_diff_on 𝕜 s) (hx : x ∈ s) (m : (fin 1) → E) :
+  (h : unique_diff_within_at 𝕜 s x) (m : fin 1 → E) :
   (iterated_fderiv_within 𝕜 1 f s x : ((fin 1) → E) → F) m
   = (fderiv_within 𝕜 f s x : E → F) (m 0) :=
-by { rw [iterated_fderiv_within_succ_apply_right hs hx, iterated_fderiv_within_zero_apply], refl }
+begin
+  simp only [iterated_fderiv_within_succ_apply_left, iterated_fderiv_within_zero_eq_comp,
+    (continuous_multilinear_curry_fin0 𝕜 E F).symm.comp_fderiv_within h],
+  refl
+end
 
-/-- If two functions coincide on a set `s` of unique differentiability, then their iterated
-differentials within this set coincide. -/
-lemma iterated_fderiv_within_congr {n : ℕ}
-  (hs : unique_diff_on 𝕜 s) (hL : ∀y∈s, f₁ y = f y) (hx : x ∈ s) :
-  iterated_fderiv_within 𝕜 n f₁ s x = iterated_fderiv_within 𝕜 n f s x :=
+lemma filter.eventually_eq.iterated_fderiv_within' (h : f₁ =ᶠ[𝓝[s] x] f) (ht : t ⊆ s) (n : ℕ) :
```

Comment:
Maybe weaken hypothesis `ht` to `s \in \nhds[t] x`? (also in a few other lemmas)

Reply:
I've already started splitting deriv lean on top of this branch, so I would prefer to postpone big changes till after we merge this one and split deriv.

Reply:
Oh, if this a big change, then sure, this can wait.

bors d+